admin/installer/_images.py

```diff
@@ -67,3 +120,269 @@ def _unserialize_packer_dict(serialized_packer_dict):
+        and modified.
+    """
+    build_region = field(type=unicode, mandatory=True)
+    publish_regions = pvector_field(item_type=unicode)
+    template = field(type=unicode, mandatory=True)
+    distribution = field(type=unicode, mandatory=True)
+    configuration_directory = field(type=FilePath, initial=PACKER_TEMPLATE_DIR)
+    source_ami = field(type=unicode, mandatory=True)
+    working_directory = field(type=FilePath, mandatory=True)
+
+
+@sync_performer
+def perform_packer_configure(dispatcher, intent):
+    """
+    Copy the prototype configuration files and provisioning scripts to a
+    temporary location and modify one of the configurations witht the values
+    found in ``intent``.
+    """
+    temporary_configuration_directory = intent.working_directory.child(
+        'packer_configuration'
+    )
+    temporary_configuration_directory.makedirs()
+    intent.configuration_directory.copyTo(temporary_configuration_directory)
+
+    template_name = (
+        u"template_{distribution}_{template}.json".format(
+            distribution=intent.distribution,
+            template=intent.template,
+        )
+    )
+    template_path = temporary_configuration_directory.child(
+        template_name
+    )
+
+    with template_path.open('r') as infile:
+        configuration = json.load(infile)
+
+    # If a region was specified, just build the image there and don't do
+    # any copying.
+    configuration['builders'][0]['region'] = intent.build_region
+    configuration['builders'][0]['source_ami'] = intent.source_ami
+    configuration['builders'][0]['ami_regions'] = thaw(intent.publish_regions)
+    output_template_path = template_path.temporarySibling()
+    with output_template_path.open('w') as outfile:
+        json.dump(configuration, outfile)
+    # XXX temporarySibling sets alwaysCreate = True for some reason.
+    output_template_path.alwaysCreate = False
+    return output_template_path
+
+
+class PackerBuild(PClass):
+    """
+    The attributes necessary to run ``packer build``.
+
+    # XXX This attribute should be called configuration_path
+    :ivar template: The path to a packer build configuration file.
+    :ivar reactor: The Twisted reactor object used to run the ``packer build``
+        subprocess.
+    :ivar sys_module: A ``sys`` like object with ``stdout`` and ``stderr``
+        attributes. The ``stderr`` of ``packer build`` will be written to
+        ``sys_module.stderr``.
+    """
+    template = field(type=FilePath)
+    reactor = field(mandatory=True)
+    sys_module = field(initial=sys)
+
+
+@deferred_performer
+def perform_packer_build(dispatcher, intent):
+    """
+    Run ``packer build`` using the configuration in the supplied ``intent`` and
+    parse its output.
+
+    :returns: A ``Deferred`` which fires with a dict mapping the ID of the AMI
+        published to each AWS region.
+    """
+    command = ['/opt/packer/packer', 'build',
```

Comment:
I guess it's not a concern if you only ever want to call it from jenkins. Can you just rely on $PATH to have packer? or is there a reason not to do that?